com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -1241,11 +1254,18 @@ public virtual void OnActionReceived(ActionBuffers actions)
 
             if (!actions.ContinuousActions.IsEmpty())
             {
-                m_LegacyActionCache = actions.ContinuousActions.Array;
+                Array.Copy(actions.ContinuousActions.Array,
+                    m_LegacyActionCache,
+                    actionSpec.NumContinuousActions);
             }
             else
             {
-                m_LegacyActionCache = Array.ConvertAll(actions.DiscreteActions.Array, x => (float)x);
+                var discreteArray = Array.ConvertAll(actions.DiscreteActions.Array, x => (float)x);
```

Comment:
This will still allocated a new array each step, right?

Reply:
yeah

Reply:
So why copy? I would say either
* Keep the old behavior
* or convert when writing to m_LegacyActionCache, e.g. `m_LegacyActionCache[i] = (float) actions.DiscreteActions.Array[i]`

Personally, I'd lean towards not optimizing deprecated code, and just go with what's simplest/easiest to maintain.